Fails on `nuxt typecheck`

I have a pipeline checks, getting this error on `nuxt typecheck`

```
node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:32:7 - error TS2339: Property 'host' does not exist on type 'unknown'.

32       host: _host = '',
         ~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:33:7 - error TS2339: Property 'id' does not exist on type 'unknown'.

33       id = '',
         ~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:34:7 - error TS2339: Property 'domains' does not exist on type 'unknown'.

34       domains = undefined,
         ~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:35:7 - error TS2339: Property 'ignoreDnt' does not exist on type 'unknown'.

35       ignoreDnt = true,
         ~~~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:36:7 - error TS2339: Property 'ignoreLocalhost' does not exist on type 'unknown'.

36       ignoreLocalhost: ignoreLocal = false,
         ~~~~~~~~~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:37:7 - error TS2339: Property 'autoTrack' does not exist on type 'unknown'.

37       autoTrack = true,
         ~~~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:38:7 - error TS2339: Property 'customEndpoint' does not exist on type 'unknown'.

38       customEndpoint: _customEP = undefined,
         ~~~~~~~~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:39:7 - error TS2339: Property 'version' does not exist on type 'unknown'.

39       version: _ver = 1,
         ~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:40:7 - error TS2339: Property 'useDirective' does not exist on type 'unknown'.

40       useDirective = false,
         ~~~~~~~~~~~~

node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:41:7 - error TS2339: Property 'debug' does not exist on type 'unknown'.

41       debug = false,
         ~~~~~


Found 10 errors in the same file, starting at: node_modules/.pnpm/nuxt-umami@2.5.6_vue@3.3.8/node_modules/nuxt-umami/internal/utils.ts:32


[1:55:25 PM]  ERROR  Command failed with exit code 2: npx -p vue-tsc -p typescript vue-tsc --noEmit

  at makeError (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/index3.mjs:625:11)
  at handlePromise (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/index3.mjs:1821:26)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async Object.run (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/typecheck.mjs:82:7)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1648:16)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1639:11)
  at async runMain$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1773:7) 



[1:55:25 PM]  ERROR  Command failed with exit code 2: npx -p vue-tsc -p typescript vue-tsc --noEmit
```

Package manager: pnpm v8.6.5


I tried to adjust tsconfig.json but it didn't help. The only solution I found was to manually apply type casting.
```
  const {
    umami: {
      host: _host = '',
      id = '',
      domains = undefined,
      ignoreDnt = true,
      ignoreLocalhost: ignoreLocal = false,
      autoTrack = true,
      customEndpoint: _customEP = undefined,
      version: _ver = 1,
      useDirective = false,
      debug = false,
    } = {} as any,     <<<==== typecast as any
  } = useAppConfig();
```

--- a/README.md
+++ b/README.md
@@ -184,6 +184,8 @@ You can pass a string as the event name, or an object containing a `name` proper
   * If you're debugging live sites, set `debug: true` in your config.
 * __Can I use Umami v2/Cloud?__
   * Yes. To use Umami v2, set `version: 2` in the Nuxt-Umami config.
+* __`nuxt typecheck` fails! What can I do to resolve it?__
+  * The problem could be tied to `pnpm`'s dependency hoisting. Thanks to [Jeet for discovering this](https://github.com/ijkml/nuxt-umami/issues/85#issuecomment-1868442446). Simply create a `.npmrc` file in the root of your Nuxt project and add `shamefully-hoist=true`. If that doesn't work, I'll be happy to look further into it.
 * __Welp, I am getting some CORS errors!__
   * Some adblockers like _uBlock_ and _Ghostery_ block Umami Cloud's endpoints. Try to disable your adblockers (yes, all of them). Also, double-check your config and Umami version. If all else fails, host your own Umami instance.
 * __How do I set up my own Umami instance?__

